falcon/app.py

```diff
@@ -226,6 +229,9 @@ def process_response(self, req, resp, resource, req_succeeded)
         'resp_options',
     )
 
+    req_options: RequestOptions
+    resp_options: ResponseOptions
```

Comment:
if sphinx like is we could also inline the documentation so that ides automatically pick it up. but it would require checking the docs output

Reply:
Sphinx should pick it up, I use it in other projects, not sure if it plays well with Napoleon, but I can try in a followup PR.